### PR TITLE
Add docs for parser error codes

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Add docs for error codes https://github.com/babel/babel/pull/13098#issuecomment-818289549

### CODE REVIEW

File: docs/parser.md [hunk 1]
```diff
@@ -273,6 +273,28 @@ You should enable these features only if you are using an older version.
...
+
+Example of using error codes with `errorRecovery`:
+
+```js
+const { parse } = require("@babel/parser");
+
+const ast = parse(`a b`, { errorRecovery: true });
+
+console.log(ast.errors[0].code); // BABEL_PARSER_SYNTAX_ERROR
+console.log(ast.errors[0].reasoncCode); // BABEL_PARSER_SOURCETYPE_MODULE_REQUIRED
```
Comment (nicolo-ribaudo) [inline docs/parser.md:295]:
```suggestion
console.log(ast.errors[0].reasoncCode); // MissingSemicolon
```

File: docs/parser.md [hunk 2]
```diff
@@ -273,6 +273,28 @@ You should enable these features only if you are using an older version.
     Some code has different meaning in Flow and in vanilla JavaScript. For example, `foo<T>(x)` is parsed as a call expression with a type argument in Flow, but as a comparison (`foo < T > x`) accordingly to the ECMAScript specification. By default, `babel-parser` parses those ambiguous constructs as Flow types only if the file starts with a `// @flow` pragma.
     Set this option to `true` to always parse files as if `// @flow` was specified.
 
+### Error codes
+
+Error codes have been introduced in `7.14.0`. This is useful if you want to handle the errors thrown by babel-parser by type.
```
Comment (nicolo-ribaudo) [inline docs/parser.md:278]: We can use an history table to say when they have been introduced. Also `babel-parser` -> `@babel/parser` (and it should be quoted with `)

File: docs/parser.md [hunk 3]
```diff
@@ -273,6 +273,35 @@ You should enable these features only if you are using an older version.
...
+### Error codes
+
+<details>
+  <summary>History</summary>
+| Version | Changes |
+| --- | --- |
+| `v7.14.0` | Added error codes |
+</details>
+
+Error codes have been introduced in `7.14.0`. This is useful if you want to handle the errors thrown by `@babel/parser` by type.
```
Comment (existentialism) [inline docs/parser.md:285]:
```suggestion
Error codes were introduced in `7.14.0`. They are useful if you want to handle the errors thrown by `@babel/parser`.
```

File: docs/parser.md [hunk 4]
```diff
@@ -273,6 +273,35 @@ You should enable these features only if you are using an older version.
...
+
+There are two error codes, `code` and `reasonCode`.
+
+- `code`
+  - Rough classification of errors(e.g.`BABEL_PARSER_SYNTAX_ERROR`,`BABEL_PARSER_SOURCETYPE_MODULE_REQUIRED`).
```
Comment (existentialism) [inline docs/parser.md:290]:
```suggestion
  - Rough classification of errors (e.g. `BABEL_PARSER_SYNTAX_ERROR`, `BABEL_PARSER_SOURCETYPE_MODULE_REQUIRED`).
```

File: docs/parser.md [hunk 5]
```diff
@@ -273,6 +273,35 @@ You should enable these features only if you are using an older version.
...
+- `reasonCode`
+  - Detailed classification of errors(e.g.`MissingSemicolon`,`VarRedeclaration`).
```
Comment (existentialism) [inline docs/parser.md:292]:
```suggestion
  - Detailed classification of errors (e.g. `MissingSemicolon`, `VarRedeclaration`).
```

File: docs/parser.md [hunk 6]
```diff
@@ -273,6 +273,35 @@ You should enable these features only if you are using an older version.
...
+
+Example of using error codes with `errorRecovery`:
+
+```js
+const { parse } = require("@babel/parser");
+
+const ast = parse(`a b`, { errorRecovery: true });
+
+console.log(ast.errors[0].code); // BABEL_PARSER_SYNTAX_ERROR
+console.log(ast.errors[0].reasoncCode); // MissingSemicolon
```
Comment (existentialism) [inline docs/parser.md:302]:
```suggestion
console.log(ast.errors[0].reasonCode); // MissingSemicolon
```

**Review by existentialism — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

**Pull request merged**